File: src/Pidget/rcrm.pidgets.multi_user_diary/src/Models/TaskInfo.ts
import { TaskAssignmentInfo } from './TaskAssignmentInfo';
import { TaskRecurrenceInfo } from './TaskRecurrenceInfo';

export type TaskInfo = {
    taskId: number;
    subject: string;
    dueDate: Date;
    startDate: Date;
    statusDescription: string;
    priorityDescription: string;
    percCompleted?: number;
    completedDate?: Date;
    isPrivate?: string;
    notebookItemId?: number;
    jobId?: number;
    personId?: number;
    clientId?: number;
    placementId?: number;
    applicantActionId?: number;
    invoiceId?: number;
    reviewListId?: number;
    mailshotId?: number;
    linkXML?: string;
    taskAssignment?: TaskAssignmentInfo;
    notes?: string;
    recurrenceInfo?: TaskRecurrenceInfo;
};

File: src/API/Models/TaskInfo.cs
namespace ARCRM.Pidgets.MultiUserDiaryAPI.Models
{
    public class TaskInfo
    {
        public int? TaskId { get; set; }

        public string? Subject { get; set; }

        public DateTime? DueDate { get; set; }

        public DateTime? StartDate { get; set; }

        public int? StatusValueId { get; set; }

        public string? StatusSystemCode { get; set; }

        public int? PriorityValueId { get; set; }

        public string? PrioritySystemCode { get; set; }

        public int? PercCompleted { get; set; }

        public DateTime? CompletedDate { get; set; }

        public string? IsPrivate { get; set; }

        public int? NotebookItemId { get; set; }

        public int? JobId { get; set; }

        public int? PersonId { get; set; }

        public int? ClientId { get; set; }

        public int? PlacementId { get; set; }

        public int? ApplicantActionId { get; set; }

        public int? InvoiceId { get; set; }

        public int? ReviewListId { get; set; }

        public int? MailshotId { get; set; }

        public string? LinkXML { get; set; }

        public string? StatusDescription { get; set; }

        public string? PriorityDescription { get; set; }

        public object? UpdatedTimestamp { get; set; }

        public TaskAssignmentInfo? TaskAssignment { get; set; }

        public string? Notes { get; set; }

        public TaskRecurrenceInfo? RecurrenceInfo { get; set; }
    }
}
